File: C++/source/Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager2.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager2.h"
#include "Regression/RegressionTestMessage.h"
#include "Framework/Messaging/LightHouse/Test/LightHouseTestContext.h"
#include "Framework/Messaging/LightHouse/Test/LightHouseTestLightPulse1.h"
#include "Framework/Utils/StringUtils.h"

namespace WaveNs
{

LightHouseTestObjectManager2::LightHouseTestObjectManager2 ()
    : WaveLocalObjectManager (getServiceName ())
{
    addLightPulseType (LightHouseTestLightPulse1::getLightPulseName ());
}

LightHouseTestObjectManager2::~LightHouseTestObjectManager2 ()
{
}

LightPulse *LightHouseTestObjectManager2::createLightPulseInstance (const string &lightPulseName)
{
    LightPulse *pLightPulse = NULL;

    if (lightPulseName == (LightHouseTestLightPulse1::getLightPulseName ()))
    {
        pLightPulse = new LightHouseTestLightPulse1 ();
    }
    else
    {
        trace (TRACE_LEVEL_FATAL, "LightHouseTestObjectManager2::createLightPulseInstance : Unknown Light Pulse Type : " + lightPulseName);
    }

    return (pLightPulse);
}

string LightHouseTestObjectManager2::getServiceName ()
{
    return ("Light House Test 2");
}

LightHouseTestObjectManager2 *LightHouseTestObjectManager2::getInstance ()
{
    static LightHouseTestObjectManager2 *pLightHouseTestObjectManager2 = new LightHouseTestObjectManager2 ();

    WaveNs::waveAssert (NULL != pLightHouseTestObjectManager2, __FILE__, __LINE__);

    return (pLightHouseTestObjectManager2);
}

WaveServiceId LightHouseTestObjectManager2::getWaveServiceId ()
{
    return ((getInstance ())->getServiceId ());
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager2.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse.Test;

import com.CxWave.Wave.Framework.ObjectModel.WaveLocalObjectManager;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.ObjectManagerPriority;
import com.CxWave.Wave.Framework.Type.WaveServiceId;
import com.CxWave.Wave.Framework.Utils.Assert.WaveAssertUtils;
import com.CxWave.Wave.Resources.ResourceEnums.WaveObjectManagerPriority;

@ObjectManagerPriority (WaveObjectManagerPriority.WAVE_OBJECT_MANAGER_PRIORITY_LIGHT_HOUSE_TEST_2)
public class LightHouseTestObjectManager2 extends WaveLocalObjectManager
{
    private static LightHouseTestObjectManager2 s_LightHouseTestObjectManager2 = null;

    public LightHouseTestObjectManager2 ()
    {
        super (getServiceName ());

        addLightPulseType (LightHouseTestLightPulse1.getLightPulseName ());
    }

    private static String getServiceName ()
    {
        return ("Light House Test 2");
    }

    public static LightHouseTestObjectManager2 getInstance ()
    {
        if (null == s_LightHouseTestObjectManager2)
        {
            s_LightHouseTestObjectManager2 = new LightHouseTestObjectManager2 ();
        }

        WaveAssertUtils.waveAssert (null != s_LightHouseTestObjectManager2);

        return (s_LightHouseTestObjectManager2);
    }

    public static WaveServiceId getWaveServiceId ()
    {
        return ((getInstance ()).getServiceId ());
    }
}
